### PR TITLE
fix!: `ConceptMapping.coding.code` should be a CURIE

### LINKED ISSUE #383: Use preferred formats for `MappableConcept.mappings`
### Feature description

Use the preferred `Coding` format for each item in `MappableConcept.mappings`. A `Coding` consists of a `relation` and `coding`, where a [Coding](https://github.com/ga4gh/gks-core/blob/1.0.0-snapshot.2024-11.2/schema/gks-core/json/Coding) requires: 
  - `code`: "A symbol uniquely identifying the concept, as in a syntax defined by the code system. CURIE format is preferred where possible (e.g. 'SO:0000704' is the CURIE form of the Sequence Ontology code for 'gene')."
  - `system`: description": "The terminology/code system that defined the code. May be reported as a free-text name (e.g. 'Sequence Ontology'), but it is preferable to provide a uri/url for the system. When the 'code' is reported as a CURIE, the 'system' should be reported as the uri that the CURIE's prefix expands to (e.g. 'http://purl.obofoundry.org/so.owl/' for the Sequence Ontology)."

### Use case

The `Coding` object allows for flexibility in how `system` and `code` are represented, which can lead to inconsistency across different implementations and systems. Aligning these fields to use preferred formats can help ensure consistent representation.

### Acceptance Criteria

Given a Mappable Concept (gene) is queried,
When it has `mappings`,
Then the `system` must be represented as a URI/URL and the `code` must be represented as a CURIE, where possible

### Proposed solution

_No response_

### Alternatives considered

_No response_

### Implementation details

_No response_

### Potential Impact

_No response_

### Additional context

_No response_

### Contribution

Yes, I can create a PR for this feature.

### PR DESCRIPTION
**Pull request by korikuzma**
close #383

* Use `concept_id` as the CURIE

### CODE REVIEW

File: src/gene/query.py [hunk 1]
```diff
@@ -424,10 +424,9 @@ def _create_concept_mapping(
                 raise ValueError(err_msg) from e
 
             system = NAMESPACE_TO_SYSTEM_URI.get(source, source)
-            code_ = concept_id.upper() if source == NamespacePrefix.HGNC else source_id
 
             return ConceptMapping(
-                coding=Coding(code=code(code_), system=system), relation=relation
+                coding=Coding(code=code(concept_id), system=system), relation=relation
```
Comment (korikuzma) [inline src/gene/query.py:429]: @jsstevenson is it fine to just use the `concept_id` as the CURIE? Technically, HGNC uses uppercase. I might be overthinking things with mappings

Comment (korikuzma) [reply src/gene/query.py:429]: Oh, maybe we just need to look at updating our `NamespacePrefix` member values in the future?

Comment (jsstevenson) [reply src/gene/query.py:429]: Yeah, I think that'd be simplest. 

Realistically there is cannot be a "true" CURIE prefix for a given source (identifiers.org was supposed to be the solution, but then it has its own problems so people made the bioregistry, but it has problems and so on and so on), which is why Matt was talking about some kind of local prefix expansions document. Some sources obviously have their own personal opinions about what prefixes you use for them, but then it begs the question of what to do if there's a community standard which differs from that. We should definitely look at updating our own prefix mapping where it makes sense though.

Comment (korikuzma) [reply src/gene/query.py:429]: @jsstevenson Ah okay, maybe we wait for Matt (and others) to provide guidance then and update our `NamespacePrefix` values whenever that discussion happens?

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**